third_party/CMakeLists.txt

```diff
@@ -10,9 +10,14 @@ target_compile_definitions(
 )
 target_include_directories(
         nlohmann_json
-        INTERFACE
+        SYSTEM INTERFACE
         $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/nlohmann>
 )
+set_source_files_properties(
+  $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/nlohmann>/json.hpp
+  TARGET_DIRECTORY nlohmann_json
+  PROPERTIES COMPILE_FLAGS -Wno-deprecated-declarations
+)
```

Comment:
Somehow I can't figure out how to suppress this warning?

```
In file included from /Users/julien/Software/Others/EnergyPlus/src/EnergyPlus/FileSystem.hh:56:
/Users/julien/Software/Others/EnergyPlus/third_party/nlohmann/json.hpp:5350:32: warning: 'char_traits<unsigned char>' is deprecated: char_traits<T> for T not equal to char, wchar_t, char8_t, char16_t or char32_t is non-standard and is provided for a temporary period. It will be removed in LLVM 18, so please migrate off of it. [-Wdeprecated-declarations]
 5350 |             auto result = std::char_traits<char_type>::to_int_type(*current);
      |                                ^
/Users/julien/Software/Others/EnergyPlus/third_party/nlohmann/json.hpp:10489:29: note: in instantiation of member function 'nlohmann::detail::iterator_input_adapter<const unsigned char *>::get_character' requested here
 10489 |         return current = ia.get_character();
       |                             ^
/Users/julien/Software/Others/EnergyPlus/third_party/nlohmann/json.hpp:8308:17: note: in instantiation of member function 'nlohmann::detail::binary_reader<nlohmann::basic_json<>, nlohmann::detail::iterator_input_adapter<const unsigned char *>>::get' requested here
 8308 |                 get();
```